mat64/matrix.go

```diff
@@ -395,23 +401,23 @@ func MaybeFloat(fn FloatPanicker) (f float64, err error) {
 }
 
 // Type Error represents matrix handling errors. These errors can be recovered by Maybe wrappers.
-type Error string
-
-func (err Error) Error() string { return string(err) }
-
-const (
-	ErrIndexOutOfRange = Error("mat64: index out of range")
-	ErrRowAccess       = Error("mat64: row index out of range")
-	ErrColAccess       = Error("mat64: column index out of range")
-	ErrZeroLength      = Error("mat64: zero length in matrix definition")
-	ErrRowLength       = Error("mat64: row length mismatch")
-	ErrColLength       = Error("mat64: col length mismatch")
-	ErrSquare          = Error("mat64: expect square matrix")
-	ErrNormOrder       = Error("mat64: invalid norm order for matrix")
-	ErrSingular        = Error("mat64: matrix is singular")
-	ErrShape           = Error("mat64: dimension mismatch")
-	ErrIllegalStride   = Error("mat64: illegal stride")
-	ErrPivot           = Error("mat64: malformed pivot list")
+type Error struct{ string }
```

Comment:
It allows us to own the mat64.Error type. With this change, it is not possible to another package to create a new non-zero Error.